utility/mm2_crawler

```diff
@@ -1026,12 +1068,28 @@ def per_host(session, host, options, config):
                 % (host.name, host.id))
     hoststate.close()
 
+    if all_categories_have_failed:
+	# Let's say that '5' is the signal for the calling function
+        # that all categories have failed due to broken base URLs
+        # and that this host should me marked as failed during crawl
+        return 5
+
     if rc == 0:
         if len(host_category_dirs) > 0:
             sync_hcds(session, host, host_category_dirs)
     return rc
 
+def count_crawl_failures(host, config):
+   try:
+       host.crawl_failures += 1
+   except TypeError:
+       host.crawl_failures = 1
 
+   auto_disable = config.get('CRAWLER_AUTO_DISABLE', 5)
+   if host.crawl_failures >= auto_disable:
+       host.disable_reason = "Host has been disabled (user_active) after %d consecutive crawl failures" % auto_disable
+       host.user_active = False
```

Comment:
Discussing this with @ralphbean we thought that it would be nice to have a way to notify the mirror admin about this. One way would be to rely on fedmsg and FMN to notify the mailing list. What do you think?